GamaGameHub/GamaGameHub/Controllers/GameController.cs

```diff
@@ -1,14 +1,35 @@
-﻿using Microsoft.AspNetCore.Authorization;
+﻿using GamaGameHub.Core.Contracts;
+using GamaGameHub.Core.Models.Pagination;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 
 namespace GamaGameHub.Controllers
 {
     [Authorize]
     public class GameController : Controller
     {
-        public GameController()
+        private readonly IGameService gameService;
+        public GameController(IGameService gameService)
         {
-            
+            this.gameService = gameService;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> Index(int page = 1)
+        {
+            var games = await this.gameService.GetGames();
+
+            if (page < 1) { page = 1; }
+
+            int totalItems = games.Count();
+            Pager pager = new Pager(totalItems, page);
+            int skipGames = (page - 1) * pager.PageSize;
+
+            var data = games.Skip(skipGames).Take(pager.PageSize).ToList();

```

Comment:
This does not look right. I think the logic for skipping and taking a certain amount of games should be in the service in order not to execute a "select * from games", but to issue a query only for the necessary games directly to the db.